### PR TITLE
Add Overlay component

### LINKED ISSUE #32: Overlay
We need to create a component that will provide an "overlay" piece of UI that will allow modals and other content to sit on top of it while still making the UI underneath the `Overlay` slightly visible.

### PR DESCRIPTION
**Pull request by stevemdixon**
### Purpose
This PR resolves #32 by adding an `Overlay` component that we will use to compose UI that is designed to appear to "sit above" other UI i.e. modals.
### Things to Note
`Overlay` has an `onClick` prop that can be used as part of a "close" UX.
### Review
Keen to see how this feels for y'all! 👍

### CODE REVIEW

File: Overlay/style.css [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+.overlay {
+  position: fixed;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  background-color: var(--outer-space-ultra-dark);
+  overflow-x: hidden;
+  z-index: 25;
```
Comment (hharnisc) [inline Overlay/style.css]: I think we need to keep these z-indexes in a file with variables 😄, that way we can see all z-index's in one place

I've seen the concept of `layers` work pretty well here

```
0000s: (optional)navbar
1000s: overlays
2000s: modals
3000s: tooltips
```

File: Overlay/style.css [hunk 2]
```diff
@@ -0,0 +1,12 @@
+@import '../variables.css';
+
+.overlay {
+  position: fixed;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  background-color: var(--outer-space-ultra-dark);
```
Comment (hharnisc) [inline Overlay/style.css]: Curious if we would want this to be the background for a tooltip as well?

File: Overlay/style.css [hunk 3]
```diff
@@ -0,0 +1,12 @@
+@import '../variables.css';
+
+.overlay {
+  position: fixed;
```
Comment (hharnisc) [inline Overlay/style.css]: I think this would be better to be positioned absolutely

```
position: absolute;
top: 0;
left: 0;
bottom: 0;
right: 0;
```

I'm thinking of the case where this is the backdrop for a tooltip and someone scrolls. That clickable layer to hide the tooltip is going to scroll off the screen.

File: Overlay/style.css (hunk 2 above)
Comment (hharnisc) [inline Overlay/style.css]: This could be a good config option, like `darkenedOverlay` or some other name?

File: Overlay/style.css [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  background-color: var(--outer-space-ultra-dark);
+  overflow-x: hidden;
```
Comment (hharnisc) [inline Overlay/style.css]: Might not need this with absolute positioning as well.

File: Overlay/style.css (hunk 3 above)
Comment (stevemdixon) [inline Overlay/style.css]: Awesome, love this idea. 👍

File: Overlay/style.css (hunk 2 above)
Comment (stevemdixon) [inline Overlay/style.css]: Hm, yeah! I think this would suit being the `background-color` for our `Tooltip` component we'll make. 😄

Comment (stevemdixon) [inline Overlay/style.css]: Curious about `darkenedOverlay`; might this be a new name for our `--outer-space-ultra-dark` variable?

File: Overlay/style.css (hunk 4 above)
Comment (stevemdixon) [inline Overlay/style.css]: Yup, good catch. So, this triggers a scroll effect similar to Trello when we open a card and can scroll it. Reflecting on this approach, that functionality feels like it's best suitable for a component implementation that uses `Overlay` rather than the implementation inheriting this behaviour. 😄 Will get this one removed!

File: Overlay/style.css (hunk 1 above)
Comment (stevemdixon) [inline Overlay/style.css]: Yup! Feels great! Will do that in another PR. 👍

**Comment by stevemdixon** [comment]
Hey, @hharnisc! 👋 No rush on this one, curious if this might be good to merge. 😁

**Comment by stevemdixon** [comment]
Hey, @hharnisc! 👋 I've gone ahead and merged `master` into this branch so that I can grab all of our new `z-index` variables and switch our the `z-index` property value of the `overlay` class to use one of these new variables. 😄
